opendatahub-io/data-science-pipelines-operator
# Add Unit Tests

## The issue resolved by this Pull Request:
Resolves #162

## Description of your changes:
- Implements the concept of 'unit' and 'func' test classes
- Adds 2 new Make commands: `make functest` and `make unittest`, which run their respective classes of tests using build labels denoted below
- `make test` still runs both unit and functional tests.
- Updates suite_test.go to run as a functional-class test suite (apply build label `test_functional`)
- Add unit tests for all sub-reconciliation functions (build label `test_unit`)


## Testing instructions
- run `make test`, ensure all passing
- run `make functest`, ensure passes
- run `make unittest`, ensure passes and >20% code coverage (specified as minimum target in issue spec)

## Checklist
